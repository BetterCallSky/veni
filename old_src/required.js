import _makeError from './_makeError';

export default (msg = 'is required') => (data) => {
  if (data.value == null) {
    return {
      stop: true,
      error: _makeError(data, msg),
    };
  }
  return undefined;
};
