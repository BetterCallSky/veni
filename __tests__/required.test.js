import required from '../src/required';

describe('required', () => {
  [undefined, null].forEach(val => {
    it(`should return an error for: ${val}`, () => {
      expect(required()({data: val})).toMatchSnapshot();
    });
  });

  it('should return a custom message', () => {
    expect(required('custom message')({data: null})).toMatchSnapshot();
  });
});
