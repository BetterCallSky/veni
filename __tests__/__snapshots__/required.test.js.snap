// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`required should return a custom message 1`] = `
Object {
  "error": Object {
    "error": "custom message",
    "path": "value",
  },
  "stop": true,
}
`;

exports[`required should return an error for: null 1`] = `
Object {
  "error": Object {
    "error": "is required",
    "path": "value",
  },
  "stop": true,
}
`;

exports[`required should return an error for: undefined 1`] = `
Object {
  "error": Object {
    "error": "is required",
    "path": "value",
  },
  "stop": true,
}
`;
